packages/google_sign_in/google_sign_in/example/lib/main.dart

```diff
@@ -7,17 +7,21 @@
 import 'dart:async';
 import 'dart:convert' show json;
 
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:google_sign_in/google_sign_in.dart';
+import 'package:google_sign_in_platform_interface/google_sign_in_platform_interface.dart';
+import 'package:google_sign_in_web/google_sign_in_web.dart' as web;
 import 'package:http/http.dart' as http;
 
+const List<String> scopes = <String>[
+  'https://www.googleapis.com/auth/contacts.readonly',
+];
+
 GoogleSignIn _googleSignIn = GoogleSignIn(
   // Optional clientId
   // clientId: '479882132969-9i9aqik3jfjd7qhci1nqf0bm2g71rm1u.apps.googleusercontent.com',
-  scopes: <String>[
-    'email',
-    'https://www.googleapis.com/auth/contacts.readonly',
-  ],
+  scopes: scopes,
 );
 
```

Comment:
I had to change a few things here to make it work for web also

```
GoogleSignIn _googleSignIn = GoogleSignIn(
  // Optional clientId
  clientId: clientId,
  scopes: scopes,
);
web.GoogleSignInPlugin _plugin =
    GoogleSignInPlatform.instance as web.GoogleSignInPlugin;

void main() async {
  await _plugin.initWithParams(
      const SignInInitParameters(scopes: scopes, clientId: clientId));
```

I'm not sure if this is the correct way since now I'm initialising the `google_sign_in_platform` and the `google_sign_in` part which feels kinda duplicated but the app throws an error that `GoogleSignInPlugin:init()` needs to be called before everything else (like it's done in the web example).

Reply:
was able to replace the _googleSignIn part compleatly using _plugin see next comments. Full example also [here](https://gist.github.com/c-seeger/6a36bd2f5561a763046d8ad254df552b)